crates/protocol/src/types/len_vec.rs

```diff
@@ -0,0 +1,107 @@
+//! A length-prefixed array (prefixed by a [`super::VarInt`])
+
+use bytes::{Buf, BufMut};
+
+use crate::{error, ProtocolType};
+
+/// There are 2 implementations of this:
+///
+///   - One for anything that's [`ProtocolType`] and can of be any length
+///   - One just for bytes (faster than decoding [`super::Primitive<u8>`])
```

Comment:
No need for extra nesting in MD:

```suggestion
/// - One for anything that's [`ProtocolType`] and can of be any length
/// - One just for bytes (faster than decoding [`super::Primitive<u8>`])
```

Reply:
Done